File: old/tags/2.2/cpp/DIS/DesignatorPdu.cpp
#include <DIS/DesignatorPdu.h> 

using namespace DIS;


DesignatorPdu::DesignatorPdu() : DistributedEmissionsFamilyPdu(),
   _designatingEntityID(), 
   _codeName(0), 
   _designatedEntityID(), 
   _designatorCode(0), 
   _designatorPower(0.0), 
   _designatorWavelength(0.0), 
   _designatorSpotWrtDesignated(), 
   _designatorSpotLocation(), 
   _deadReckoningAlgorithm(0), 
   _padding1(0), 
   _padding2(0), 
   _entityLinearAcceleration()
{
    setPduType( 24 );
}

DesignatorPdu::~DesignatorPdu()
{
}

EntityID& DesignatorPdu::getDesignatingEntityID() 
{
    return _designatingEntityID;
}

const EntityID& DesignatorPdu::getDesignatingEntityID() const
{
    return _designatingEntityID;
}

void DesignatorPdu::setDesignatingEntityID(const EntityID &pX)
{
    _designatingEntityID = pX;
}

unsigned short DesignatorPdu::getCodeName() const
{
    return _codeName;
}

void DesignatorPdu::setCodeName(unsigned short pX)
{
    _codeName = pX;
}

EntityID& DesignatorPdu::getDesignatedEntityID() 
{
    return _designatedEntityID;
}

const EntityID& DesignatorPdu::getDesignatedEntityID() const
{
    return _designatedEntityID;
}

void DesignatorPdu::setDesignatedEntityID(const EntityID &pX)
{
    _designatedEntityID = pX;
}

unsigned short DesignatorPdu::getDesignatorCode() const
{
    return _designatorCode;
}

void DesignatorPdu::setDesignatorCode(unsigned short pX)
{
    _designatorCode = pX;
}

float DesignatorPdu::getDesignatorPower() const
{
    return _designatorPower;
}

void DesignatorPdu::setDesignatorPower(float pX)
{
    _designatorPower = pX;
}

float DesignatorPdu::getDesignatorWavelength() const
{
    return _designatorWavelength;
}

void DesignatorPdu::setDesignatorWavelength(float pX)
{
    _designatorWavelength = pX;
}

Vector3Float& DesignatorPdu::getDesignatorSpotWrtDesignated() 
{
    return _designatorSpotWrtDesignated;
}

const Vector3Float& DesignatorPdu::getDesignatorSpotWrtDesignated() const
{
    return _designatorSpotWrtDesignated;
}

void DesignatorPdu::setDesignatorSpotWrtDesignated(const Vector3Float &pX)
{
    _designatorSpotWrtDesignated = pX;
}

Vector3Double& DesignatorPdu::getDesignatorSpotLocation() 
{
    return _designatorSpotLocation;
}

const Vector3Double& DesignatorPdu::getDesignatorSpotLocation() const
{
    return _designatorSpotLocation;
}

void DesignatorPdu::setDesignatorSpotLocation(const Vector3Double &pX)
{
    _designatorSpotLocation = pX;
}

char DesignatorPdu::getDeadReckoningAlgorithm() const
{
    return _deadReckoningAlgorithm;
}

void DesignatorPdu::setDeadReckoningAlgorithm(char pX)
{
    _deadReckoningAlgorithm = pX;
}

unsigned short DesignatorPdu::getPadding1() const
{
    return _padding1;
}

void DesignatorPdu::setPadding1(unsigned short pX)
{
    _padding1 = pX;
}

char DesignatorPdu::getPadding2() const
{
    return _padding2;
}

void DesignatorPdu::setPadding2(char pX)
{
    _padding2 = pX;
}

Vector3Float& DesignatorPdu::getEntityLinearAcceleration() 
{
    return _entityLinearAcceleration;
}

const Vector3Float& DesignatorPdu::getEntityLinearAcceleration() const
{
    return _entityLinearAcceleration;
}

void DesignatorPdu::setEntityLinearAcceleration(const Vector3Float &pX)
{
    _entityLinearAcceleration = pX;
}

void DesignatorPdu::marshal(DataStream& dataStream) const
{
    DistributedEmissionsFamilyPdu::marshal(dataStream); // Marshal information in superclass first
    _designatingEntityID.marshal(dataStream);
    dataStream << _codeName;
    _designatedEntityID.marshal(dataStream);
    dataStream << _designatorCode;
    dataStream << _designatorPower;
    dataStream << _designatorWavelength;
    _designatorSpotWrtDesignated.marshal(dataStream);
    _designatorSpotLocation.marshal(dataStream);
    dataStream << _deadReckoningAlgorithm;
    dataStream << _padding1;
    dataStream << _padding2;
    _entityLinearAcceleration.marshal(dataStream);
}

void DesignatorPdu::unmarshal(DataStream& dataStream)
{
    DistributedEmissionsFamilyPdu::unmarshal(dataStream); // unmarshal information in superclass first
    _designatingEntityID.unmarshal(dataStream);
    dataStream >> _codeName;
    _designatedEntityID.unmarshal(dataStream);
    dataStream >> _designatorCode;
    dataStream >> _designatorPower;
    dataStream >> _designatorWavelength;
    _designatorSpotWrtDesignated.unmarshal(dataStream);
    _designatorSpotLocation.unmarshal(dataStream);
    dataStream >> _deadReckoningAlgorithm;
    dataStream >> _padding1;
    dataStream >> _padding2;
    _entityLinearAcceleration.unmarshal(dataStream);
}


bool DesignatorPdu::operator ==(const DesignatorPdu& rhs) const
 {
     bool ivarsEqual = true;

     ivarsEqual = DistributedEmissionsFamilyPdu::operator==(rhs);

     if( ! (_designatingEntityID == rhs._designatingEntityID) ) ivarsEqual = false;
     if( ! (_codeName == rhs._codeName) ) ivarsEqual = false;
     if( ! (_designatedEntityID == rhs._designatedEntityID) ) ivarsEqual = false;
     if( ! (_designatorCode == rhs._designatorCode) ) ivarsEqual = false;
     if( ! (_designatorPower == rhs._designatorPower) ) ivarsEqual = false;
     if( ! (_designatorWavelength == rhs._designatorWavelength) ) ivarsEqual = false;
     if( ! (_designatorSpotWrtDesignated == rhs._designatorSpotWrtDesignated) ) ivarsEqual = false;
     if( ! (_designatorSpotLocation == rhs._designatorSpotLocation) ) ivarsEqual = false;
     if( ! (_deadReckoningAlgorithm == rhs._deadReckoningAlgorithm) ) ivarsEqual = false;
     if( ! (_padding1 == rhs._padding1) ) ivarsEqual = false;
     if( ! (_padding2 == rhs._padding2) ) ivarsEqual = false;
     if( ! (_entityLinearAcceleration == rhs._entityLinearAcceleration) ) ivarsEqual = false;

    return ivarsEqual;
 }

int DesignatorPdu::getMarshalledSize() const
{
   int marshalSize = 0;

   marshalSize = DistributedEmissionsFamilyPdu::getMarshalledSize();
   marshalSize = marshalSize + _designatingEntityID.getMarshalledSize();  // _designatingEntityID
   marshalSize = marshalSize + 2;  // _codeName
   marshalSize = marshalSize + _designatedEntityID.getMarshalledSize();  // _designatedEntityID
   marshalSize = marshalSize + 2;  // _designatorCode
   marshalSize = marshalSize + 4;  // _designatorPower
   marshalSize = marshalSize + 4;  // _designatorWavelength
   marshalSize = marshalSize + _designatorSpotWrtDesignated.getMarshalledSize();  // _designatorSpotWrtDesignated
   marshalSize = marshalSize + _designatorSpotLocation.getMarshalledSize();  // _designatorSpotLocation
   marshalSize = marshalSize + 1;  // _deadReckoningAlgorithm
   marshalSize = marshalSize + 2;  // _padding1
   marshalSize = marshalSize + 1;  // _padding2
   marshalSize = marshalSize + _entityLinearAcceleration.getMarshalledSize();  // _entityLinearAcceleration
    return marshalSize;
}

File: old/tags/2.1/java/edu/nps/moves/dis/DesignatorPdu.java
package edu.nps.moves.dis;

import java.util.*;
import java.io.*;
import edu.nps.moves.jaxb.dis.*;

/**
 * Section 5.3.7.2. Handles designating operations. COMPLETE
 *
 * Copyright (c) 2007, MOVES Institute, Naval Postgraduate School. All rights reserved.
 * This work is licensed under the BSD open source license, available at https://www.movesinstitute.org/licenses/bsd.html
 *
 * @author DMcG
 */
public class DesignatorPdu extends DistributedEmissionsFamilyPdu implements Serializable
{
   /** ID of the entity designating */
   protected EntityID  designatingEntityID = new EntityID(); 

   /** This field shall specify a unique emitter database number assigned to  differentiate between otherwise similar or identical emitter beams within an emitter system. */
   protected int  codeName;

   /** ID of the entity being designated */
   protected EntityID  designatedEntityID = new EntityID(); 

   /** This field shall identify the designator code being used by the designating entity  */
   protected int  designatorCode;

   /** This field shall identify the designator output power in watts */
   protected float  designatorPower;

   /** This field shall identify the designator wavelength in units of microns */
   protected float  designatorWavelength;

   /** designtor spot wrt the designated entity */
   protected Vector3Float  designatorSpotWrtDesignated = new Vector3Float(); 

   /** designtor spot wrt the designated entity */
   protected Vector3Double  designatorSpotLocation = new Vector3Double(); 

   /** Dead reckoning algorithm */
   protected byte  deadReckoningAlgorithm;

   /** padding */
   protected int  padding1;

   /** padding */
   protected byte  padding2;

   /** linear accelleration of entity */
   protected Vector3Float  entityLinearAcceleration = new Vector3Float(); 


/** Constructor */
 public DesignatorPdu()
 {
    setPduType( (short)24 );
 }

/** 
 * Constructor--takes a parallel jaxb object and returns an open-dis object 
 * 1.4_sed_bait_start */
 public DesignatorPdu(edu.nps.moves.jaxb.dis.DesignatorPdu x)
 {
     super(x); // Call superclass constructor


     edu.nps.moves.dis.EntityID foo_0;
     if(x.getDesignatingEntityID() == null)
        foo_0 = new edu.nps.moves.dis.EntityID();
      else
        foo_0 = new edu.nps.moves.dis.EntityID(x.getDesignatingEntityID() );
     this.setDesignatingEntityID(foo_0);

     this.codeName = x.getCodeName();

     edu.nps.moves.dis.EntityID foo_2;
     if(x.getDesignatedEntityID() == null)
        foo_2 = new edu.nps.moves.dis.EntityID();
      else
        foo_2 = new edu.nps.moves.dis.EntityID(x.getDesignatedEntityID() );
     this.setDesignatedEntityID(foo_2);

     this.designatorCode = x.getDesignatorCode();
     this.designatorPower = x.getDesignatorPower();
     this.designatorWavelength = x.getDesignatorWavelength();

     edu.nps.moves.dis.Vector3Float foo_6;
     if(x.getDesignatorSpotWrtDesignated() == null)
        foo_6 = new edu.nps.moves.dis.Vector3Float();
      else
        foo_6 = new edu.nps.moves.dis.Vector3Float(x.getDesignatorSpotWrtDesignated() );
     this.setDesignatorSpotWrtDesignated(foo_6);


     edu.nps.moves.dis.Vector3Double foo_7;
     if(x.getDesignatorSpotLocation() == null)
        foo_7 = new edu.nps.moves.dis.Vector3Double();
      else
        foo_7 = new edu.nps.moves.dis.Vector3Double(x.getDesignatorSpotLocation() );
     this.setDesignatorSpotLocation(foo_7);

     this.deadReckoningAlgorithm = x.getDeadReckoningAlgorithm();
     this.padding1 = x.getPadding1();
     this.padding2 = x.getPadding2();

     edu.nps.moves.dis.Vector3Float foo_11;
     if(x.getEntityLinearAcceleration() == null)
        foo_11 = new edu.nps.moves.dis.Vector3Float();
      else
        foo_11 = new edu.nps.moves.dis.Vector3Float(x.getEntityLinearAcceleration() );
     this.setEntityLinearAcceleration(foo_11);

 }
/* 1.4_sed_bait_end */


/**
 * returns a jaxb object intialized from this object, given an empty jaxb object
 * 1.4_sed_bait_start **/
 public edu.nps.moves.jaxb.dis.DesignatorPdu initializeJaxbObject(edu.nps.moves.jaxb.dis.DesignatorPdu x)
 {
     super.initializeJaxbObject(x); // Call superclass initializer

     ObjectFactory factory = new ObjectFactory();

     x.setDesignatingEntityID( this.getDesignatingEntityID().initializeJaxbObject(factory.createEntityID()) );
     x.setCodeName( this.getCodeName() );
     x.setDesignatedEntityID( this.getDesignatedEntityID().initializeJaxbObject(factory.createEntityID()) );
     x.setDesignatorCode( this.getDesignatorCode() );
     x.setDesignatorPower( this.getDesignatorPower() );
     x.setDesignatorWavelength( this.getDesignatorWavelength() );
     x.setDesignatorSpotWrtDesignated( this.getDesignatorSpotWrtDesignated().initializeJaxbObject(factory.createVector3Float()) );
     x.setDesignatorSpotLocation( this.getDesignatorSpotLocation().initializeJaxbObject(factory.createVector3Double()) );
     x.setDeadReckoningAlgorithm( this.getDeadReckoningAlgorithm() );
     x.setPadding1( this.getPadding1() );
     x.setPadding2( this.getPadding2() );
     x.setEntityLinearAcceleration( this.getEntityLinearAcceleration().initializeJaxbObject(factory.createVector3Float()) );
   return x;
 }
/* 1.4_sed_bait_end */


public int getMarshalledSize()
{
   int marshalSize = 0; 

   marshalSize = super.getMarshalledSize();
   marshalSize = marshalSize + designatingEntityID.getMarshalledSize();  // designatingEntityID
   marshalSize = marshalSize + 2;  // codeName
   marshalSize = marshalSize + designatedEntityID.getMarshalledSize();  // designatedEntityID
   marshalSize = marshalSize + 2;  // designatorCode
   marshalSize = marshalSize + 4;  // designatorPower
   marshalSize = marshalSize + 4;  // designatorWavelength
   marshalSize = marshalSize + designatorSpotWrtDesignated.getMarshalledSize();  // designatorSpotWrtDesignated
   marshalSize = marshalSize + designatorSpotLocation.getMarshalledSize();  // designatorSpotLocation
   marshalSize = marshalSize + 1;  // deadReckoningAlgorithm
   marshalSize = marshalSize + 2;  // padding1
   marshalSize = marshalSize + 1;  // padding2
   marshalSize = marshalSize + entityLinearAcceleration.getMarshalledSize();  // entityLinearAcceleration

   return marshalSize;
}


public void setDesignatingEntityID(EntityID pDesignatingEntityID)
{ designatingEntityID = pDesignatingEntityID;
}

public EntityID getDesignatingEntityID()
{ return designatingEntityID; }

public void setCodeName(int pCodeName)
{ codeName = pCodeName;
}

public int getCodeName()
{ return codeName; 
}

public void setDesignatedEntityID(EntityID pDesignatedEntityID)
{ designatedEntityID = pDesignatedEntityID;
}

public EntityID getDesignatedEntityID()
{ return designatedEntityID; }

public void setDesignatorCode(int pDesignatorCode)
{ designatorCode = pDesignatorCode;
}

public int getDesignatorCode()
{ return designatorCode; 
}

public void setDesignatorPower(float pDesignatorPower)
{ designatorPower = pDesignatorPower;
}

public float getDesignatorPower()
{ return designatorPower; 
}

public void setDesignatorWavelength(float pDesignatorWavelength)
{ designatorWavelength = pDesignatorWavelength;
}

public float getDesignatorWavelength()
{ return designatorWavelength; 
}

public void setDesignatorSpotWrtDesignated(Vector3Float pDesignatorSpotWrtDesignated)
{ designatorSpotWrtDesignated = pDesignatorSpotWrtDesignated;
}

public Vector3Float getDesignatorSpotWrtDesignated()
{ return designatorSpotWrtDesignated; }

public void setDesignatorSpotLocation(Vector3Double pDesignatorSpotLocation)
{ designatorSpotLocation = pDesignatorSpotLocation;
}

public Vector3Double getDesignatorSpotLocation()
{ return designatorSpotLocation; }

public void setDeadReckoningAlgorithm(byte pDeadReckoningAlgorithm)
{ deadReckoningAlgorithm = pDeadReckoningAlgorithm;
}

public byte getDeadReckoningAlgorithm()
{ return deadReckoningAlgorithm; 
}

public void setPadding1(int pPadding1)
{ padding1 = pPadding1;
}

public int getPadding1()
{ return padding1; 
}

public void setPadding2(byte pPadding2)
{ padding2 = pPadding2;
}

public byte getPadding2()
{ return padding2; 
}

public void setEntityLinearAcceleration(Vector3Float pEntityLinearAcceleration)
{ entityLinearAcceleration = pEntityLinearAcceleration;
}

public Vector3Float getEntityLinearAcceleration()
{ return entityLinearAcceleration; }


public void marshal(DataOutputStream dos)
{
    super.marshal(dos);
    try 
    {
       designatingEntityID.marshal(dos);
       dos.writeShort( (short)codeName);
       designatedEntityID.marshal(dos);
       dos.writeShort( (short)designatorCode);
       dos.writeFloat( (float)designatorPower);
       dos.writeFloat( (float)designatorWavelength);
       designatorSpotWrtDesignated.marshal(dos);
       designatorSpotLocation.marshal(dos);
       dos.writeByte( (byte)deadReckoningAlgorithm);
       dos.writeShort( (short)padding1);
       dos.writeByte( (byte)padding2);
       entityLinearAcceleration.marshal(dos);
    } // end try 
    catch(Exception e)
    { 
      System.out.println(e);}
    } // end of marshal method

public void unmarshal(DataInputStream dis)
{
    super.unmarshal(dis);

    try 
    {
       designatingEntityID.unmarshal(dis);
       codeName = dis.readShort();
       designatedEntityID.unmarshal(dis);
       designatorCode = dis.readShort();
       designatorPower = dis.readFloat();
       designatorWavelength = dis.readFloat();
       designatorSpotWrtDesignated.unmarshal(dis);
       designatorSpotLocation.unmarshal(dis);
       deadReckoningAlgorithm = dis.readByte();
       padding1 = dis.readShort();
       padding2 = dis.readByte();
       entityLinearAcceleration.unmarshal(dis);
    } // end try 
   catch(Exception e)
    { 
      System.out.println(e); 
    }
 } // end of unmarshal method 


 /**
  * The equals method doesn't always work--mostly on on classes that consist only of primitives. Be careful.
  */
 public boolean equals(DesignatorPdu rhs)
 {
     boolean ivarsEqual = true;

    if(rhs.getClass() != this.getClass())
        return false;

     if( ! (designatingEntityID.equals( rhs.designatingEntityID) )) ivarsEqual = false;
     if( ! (codeName == rhs.codeName)) ivarsEqual = false;
     if( ! (designatedEntityID.equals( rhs.designatedEntityID) )) ivarsEqual = false;
     if( ! (designatorCode == rhs.designatorCode)) ivarsEqual = false;
     if( ! (designatorPower == rhs.designatorPower)) ivarsEqual = false;
     if( ! (designatorWavelength == rhs.designatorWavelength)) ivarsEqual = false;
     if( ! (designatorSpotWrtDesignated.equals( rhs.designatorSpotWrtDesignated) )) ivarsEqual = false;
     if( ! (designatorSpotLocation.equals( rhs.designatorSpotLocation) )) ivarsEqual = false;
     if( ! (deadReckoningAlgorithm == rhs.deadReckoningAlgorithm)) ivarsEqual = false;
     if( ! (padding1 == rhs.padding1)) ivarsEqual = false;
     if( ! (padding2 == rhs.padding2)) ivarsEqual = false;
     if( ! (entityLinearAcceleration.equals( rhs.entityLinearAcceleration) )) ivarsEqual = false;

    return ivarsEqual;
 }
} // end of class
